src/test/kotlin/no/nav/familie/ef/sak/behandlingsflyt/steg/BeregnYtelseStegTest.kt

```diff
@@ -64,357 +65,443 @@ internal class BeregnYtelseStegTest {
+            assertThat(nyeAndeler[0].stønadFom).isEqualTo(forrigeAndelFom)
+            assertThat(nyeAndeler[0].stønadTom).isEqualTo(forrigeAndelTom)
+            assertThat(nyeAndeler[0].beløp).isEqualTo(50)
+            assertThat(nyeAndeler[1].stønadFom).isEqualTo(nyAndelFom)
+            assertThat(nyeAndeler[1].stønadTom).isEqualTo(opphør1.fradato.minusDays(1))
+            assertThat(nyeAndeler[1].beløp).isEqualTo(100)
+            assertThat(nyeAndeler[2].stønadFom).isEqualTo(opphør1.tildato.plusDays(1))
+            assertThat(nyeAndeler[2].stønadTom).isEqualTo(opphør2.fradato.minusDays(1))
+            assertThat(nyeAndeler[2].beløp).isEqualTo(100)
+            assertThat(nyeAndeler[3].stønadFom).isEqualTo(opphør2.tildato.plusDays(1))
+            assertThat(nyeAndeler[3].stønadTom).isEqualTo(opphør3.fradato.minusDays(1))
+            assertThat(nyeAndeler[3].beløp).isEqualTo(100)
+            assertThat(nyeAndeler[4].stønadFom).isEqualTo(opphør3.tildato.plusDays(1))
+            assertThat(nyeAndeler[4].stønadTom).isEqualTo(nyAndelTom)
+            assertThat(nyeAndeler[4].beløp).isEqualTo(100)
+            assertThat(nyeAndeler[0].kildeBehandlingId).isNotEqualTo(nyeAndeler[1].kildeBehandlingId)
+        }
+
+        @Test
+        internal fun `lager opphold i eksisterende andel`() {
+            val forrigeAndelFom = LocalDate.of(2021, 1, 1)
+            val forrigeAndelTom = LocalDate.of(2025, 10, 31)
+            val opphør1 = Periode(LocalDate.of(2021, 12, 1), LocalDate.of(2021, 12, 31))
+            val opphør2 = Periode(LocalDate.of(2022, 2, 1), LocalDate.of(2022, 3, 31))
+            val opphør3 = Periode(LocalDate.of(2022, 6, 1), LocalDate.of(2022, 8, 31))
+            val opphørsperioder = listOf(opphør1, opphør2, opphør3)
+            val forrigeAndeler = listOf(lagAndelTilkjentYtelse(50, forrigeAndelFom, forrigeAndelTom))
+
+            val nyeAndeler = steg.slåSammenAndelerSomSkalVidereføres(listOf(), lagTilkjentYtelse(forrigeAndeler), opphørsperioder)
+
+            assertThat(nyeAndeler).hasSize(4)
+            assertThat(nyeAndeler[0].stønadFom).isEqualTo(forrigeAndelFom)
+            assertThat(nyeAndeler[0].stønadTom).isEqualTo(opphør1.fradato.minusDays(1))
+            assertThat(nyeAndeler[0].beløp).isEqualTo(50)
+            assertThat(nyeAndeler[1].stønadFom).isEqualTo(opphør1.tildato.plusDays(1))
+            assertThat(nyeAndeler[1].stønadTom).isEqualTo(opphør2.fradato.minusDays(1))
+            assertThat(nyeAndeler[1].beløp).isEqualTo(50)
+            assertThat(nyeAndeler[2].stønadFom).isEqualTo(opphør2.tildato.plusDays(1))
+            assertThat(nyeAndeler[2].stønadTom).isEqualTo(opphør3.fradato.minusDays(1))
+            assertThat(nyeAndeler[2].beløp).isEqualTo(50)
+            assertThat(nyeAndeler[3].stønadFom).isEqualTo(opphør3.tildato.plusDays(1))
+            assertThat(nyeAndeler[3].stønadTom).isEqualTo(forrigeAndelTom)
+            assertThat(nyeAndeler[3].beløp).isEqualTo(50)
```

Comment:
Her er jeg usikker på om logikken som er implementert gir mening for saksbehandler. 
Saksbehandler får nok et "GUI" hvor de må punche inn opphør etterfulgt av nye perioder som må legges inn. Dersom de bare legger inn "opphør" for en periode så tror jeg ikke vi ønsker at den tidligere perioden videreføres etter opphøret, men at de "nye" andelene skal flettes inn i dette. 

For denne testen vil jeg egentlig mene at det kun skal gjenstå én periode: `2021-01-01 - 2021-11-30`

Reply:
Jeg så for meg at dette kunne være nyttig. Søker er tilkjent ytelse for 1. januar til 31. desember. Man ønsker å legge inn et opphold for juni og ikke noe annet. Man legger kun inn oppholdet og systemet gjør resten.